client.js

```diff
@@ -102,67 +82,98 @@ async function insert({ _, ...rest }) {
   }
 }
 
-//Requests basic information about the target node
-function summary() {
+// Requests basic information about the target node
+async function summary() {
   console.log("Client requesting summary:");
-  client.summary({ id: 1 }, (err, node) => {
-    if (err) {
-      console.log(err);
-      console.log(node);
-    } else {
-      //console.log(node);
-      console.log(
-        `The node returned id: ${node.id}, ip: ${node.ip}, port: ${node.port}`
-      );
-    }
-  });
+  try {
+    const node = await client.summary({ id: 1 });
+    console.log(
+      `The node returned id: ${node.id}, ip: ${node.ip}, port: ${node.port}`
+    );
+  } catch (err) {
+    console.error(err);
+  }
 }
 
-//Requests basic information about the target node
-async function crawl() {
-  client = new chord.Node(
-    `${lastNode.ip}:${lastNode.port}`,
-    grpc.credentials.createInsecure()
-  );
-  // The argument is total garbage
-  client.getSuccessor_remotehelper({ id: 99 }, (err, node) => {
-    if (err) {
-      console.log("Error is : ", err);
-      let nodeToDelete = lastNode.id;
-      // Remove the node from the bucket and select a random node
-      for (elem in Object.keys(bigBucketOfData)) {
-        if (
-          elem &&
-          bigBucketOfData[elem] &&
-          bigBucketOfData[elem].successor &&
-          bigBucketOfData[elem].successor.id &&
-          bigBucketOfData[elem].successor.id !== lastNode.id
-        ) {
-          lastNode = bigBucketOfData[elem].successor;
-          break;
-        }
-      }
-      delete bigBucketOfData[nodeToDelete];
-    } else {
-      if (bigBucketOfData[lastNode.id]) {
-        bigBucketOfData[lastNode.id].successor = node;
+class ChordCrawler {
+  constructor(ip, port, stepInSeconds) {
+    this.ip = ip;
+    this.port = port;
+    this.state = {};
+    this.ring = new Set([]);
+    setInterval(async () => {
+      await this.crawl();
+    }, stepInSeconds);
+  }
+  static buildConnectionString(ip, port) {
```

Comment:
Is it really cleaner to have this method instead of just concatanating in the two places that is necessary? (humble question)